src/object_storage/protocol.rs

```diff
@@ -39,6 +39,14 @@ pub struct Object {
     pub name: String,
 }
 
+pub struct ObjectCreate<R> {
```

Comment:
Logically, stuff in `protocol.rs` is raw protocol objects (JSON structures converted from official docs). This is a bit of an outlier. What I'd rather see if adding a `headers` argument to `api::create_object` with the higher layer (the one in `objects.rs`) responsible for populating it.

Reply:
I understand. I have remove the struct `ObjectCreate` in 57ba9ce03c6ff6760699cdb7f267fbad0b061f94.